sklearn/utils/fixes.py

```diff
@@ -24,11 +24,18 @@
 
 from .deprecation import deprecated
 
-try:
-    from pkg_resources import parse_version  # type: ignore
-except ImportError:
-    # setuptools not installed
-    parse_version = LooseVersion  # type: ignore
+
+def parse_version(v):
+    if not {*v} <= {*"0123456789."}:
```

Comment:
Nit: Do you think this would be clearer? (I have no strong preference)

```suggestion
    if not set(v) <= set("0123456789."):
```

Reply:
sure, works for me.